cmd/commands/errors.go

```diff
@@ -0,0 +1,59 @@
+package commands
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"strings"
+)
+
+// Joins and formats results.
+// Returns nil if all errors are nil.
+// If there is only one non-nil error, the  error is returned as-is.
+func MergeResults(input []error) error {
+	stats := computeStats(input)
+	length := stats.nilFreeSliceLength
+	if length == 0 {
+		return nil
+	}
+	if length == 1 {
+		return input[stats.firstNonNilIndex]
+	}
+	return errors.New(formatErrorMessages(input))
+}
+
+func formatErrorMessages(input []error) string {
```

Comment:
I'm not crazy about the generic error format, we may want to allow a caller to provide a template for the error message.

For example, when deleting a channel, I'd like to see an error like:

```
fmt.Sprintf("Unable to delete channel %s: %v", context, err)
```

Reply:
what would `context` be in this example?

Reply:
context == channel name, alright 😅

Reply:
Done